main.tf

```diff
@@ -499,16 +498,16 @@ data "aws_iam_policy_document" "sqs_with_kms_and_redrive_policy_document" {
       "arn:aws:kms:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:alias/${var.kms_alias}",
     ]
 
-    actions = [
-      "kms:ReEncrypt",
-      "kms:GetKeyRotationStatus",
-      "kms:GetKeyPolicy",
-      "kms:GenerateRandom",
-      "kms:GenerateDataKeyWithoutPlaintext",
-      "kms:GenerateDataKey",
-      "kms:Encrypt",
-      "kms:DescribeKey",
-      "kms:Decrypt",
-    ]
+    actions = concat([
+      "sqs:AddPermission",
+      "sqs:ChangeMessageVisibility*",
+      "sqs:DeleteMessage*",
+      "sqs:Get*",
+      "sqs:List*",
+      "sqs:PurgeQueue",
+      "sqs:ReceiveMessage",
+      "sqs:RemovePermission",
+      "sqs:Send*",
+    ], var.enable_set_attributes ? ["sqs:SetQueueAttributes"] : [])
```

Comment:
You're changing the KMS actions to something completely different. There's no need to change this.